cli-plugins/manager/candidate_test.go

```diff
@@ -70,6 +72,7 @@ func TestValidateCandidate(t *testing.T) {
 			assert.ErrorContains(t, err, tc.err)
 		} else if tc.invalid != "" {
 			assert.NilError(t, err)
+			assert.Assert(t, cmp.ErrorType(p.Err, reflect.TypeOf(&pluginError{})))
```

Comment:
Should we have an interface for plugin errors?

```go
type PluginError interface {
	Plugin()
}
```

Reply:
I'm not sure what that would buy us, only these unit tests really need to check and that's only because we want the `MarshalText` behaviour of `pluginError` (which normal `error` lacks, see the comment in `cli-plugins/manager/error.go`) for the `docker info --format '{{json .}}` case. No actual user/calling code should ever need to probe whether the `error` in `Plugin.Err` is one of these.